src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -57,7 +57,8 @@ internal sealed class VariableState
         private readonly SourceAssemblySymbol _sourceAssembly;
 
         private readonly Binder _binder;
-        private readonly Conversions _conversions;
+        private readonly Conversions _conversionsUnknownMatchesAnyTrue;
```

Comment:
>_conversionsUnknownMatchesAnyTrue [](start = 37, length = 33)

A comment would be useful here: why do we need UnknownMatchesAnyTrue in some situations, versus False in others?